src/libraries/Microsoft.PowerFx.Core/Localization/StringResources.cs

```diff
@@ -25,72 +24,193 @@ internal static class StringResources
         // If the dependency on ExternalStringResources is removed, this can be as well
         public static bool ShouldThrowIfMissing { get; set; } = true;
 
-        private static readonly ThreadSafeResouceManager _resourceManager = new ThreadSafeResouceManager();
+        internal static readonly IExternalStringResources LocalStringResources = new PowerFxStringResources("Microsoft.PowerFx.Core.strings.PowerFxResources", typeof(StringResources).Assembly, true);
 
-        [ThreadSafeProtectedByLockAttribute("_errorResources")]
-        private static readonly Dictionary<string, Dictionary<string, ErrorResource>> _errorResources = new Dictionary<string, Dictionary<string, ErrorResource>>(StringComparer.OrdinalIgnoreCase);
+        public static string Get(string resourceKey, string locale = null)
+        {
+            return TryGet(resourceKey, out string resourceValue, locale) ? resourceValue : null;
+        }
+
+        public static bool TryGet(string resourceKey, out string resourceValue, string locale = null)
+        {
+            bool success = LocalStringResources.TryGet(resourceKey, out resourceValue, locale);
+
+            if (success)
+            {
+                return true;
```

Comment:
[minor] For consistency with the next statement, use `if (LocalStringResources.TryGet(...)) { return true; }`